bin/npm-prefix.js

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env node
+// This is a single-use bin to help windows discover the proper prefix for npm
+// without having to load all of npm first
+// It does not accept argv params
+
+const path = require('path')
+const Config = require('@npmcli/config')
+const { definitions, flatten, shorthands } = require('@npmcli/config/lib/definitions')
+const config = new Config({
+  npmPath: path.dirname(__dirname),
+  // argv is explicitly not looked at since prefix is not something that can be changed via argv
+  argv: [],
+  definitions,
+  flatten,
+  shorthands,
+  excludeNpmCwd: false,
+})
+
+async function main () {
+  try {
+    await config.load()
+    // eslint-disable-next-line no-console
+    console.log(config.globalPrefix)
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.trace(err)
```

Comment:
Do you have an example of what the output looks like here in the case of an error?

I think `err` will have its own stack trace so I'm curious what it looks like with the stack from `console.trace` as well?

Reply:
console.trace:
```sh
~/D/n/scratch $ node throw.js 
Trace: Error: Hi Luke
    at t (/Users/wraithgar/Development/npm/scratch/throw.js:3:11)
    at Object.<anonymous> (/Users/wraithgar/Development/npm/scratch/throw.js:9:1)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49
    at t (/Users/wraithgar/Development/npm/scratch/throw.js:5:13)
    at Object.<anonymous> (/Users/wraithgar/Development/npm/scratch/throw.js:9:1)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49
```

console.log:
```
~/D/n/scratch $ node throw.js 
Error: Hi Luke
    at t (/Users/wraithgar/Development/npm/scratch/throw.js:3:11)
    at Object.<anonymous> (/Users/wraithgar/Development/npm/scratch/throw.js:9:1)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49
```

If I remember correctly I got in this habit because it actually logs the line in which the trace is called, not just the error. This can potentially help you track down where it's caught, not just where it's thrown.

Reply:
It's probably overkill.